src/linked_list.py

```diff
@@ -0,0 +1,96 @@
+"""The Linked List Module Creates a LinkedList."""
+
+
+class Node(object):
+    """Defining Node Class.
+    __init__() adds a val property and an optional next property
+    """
+
+    def __init__(self, val, next=None):
+        """Instantiate Node Class."""
+        self.val = val
+        self.next = next
+
+
+class LinkedList(object):
+    """Defining class of LinkedList. 
+
+    push(val): adds a node at the head, increments size of list.
+    pop(): removes the node at the head, deincrements size of list.
+    size(): returns size of LinkedList
+    search(val): searches list for node containing val and returns that node.
+    remove(node_to_delete): takes a node as argument, searches for it, and removes it from list, then deincrements size of list
+    display(): returns LinkedList as a stringified tuple-looking object.
+
+    """
+
+    def __init__(self, iterable=None):
+        """Instantiate LinkedList class."""
+        self.head = None
+        self.size_of_list = 0
+        if iterable and hasattr(iterable, "__iter__") or isinstance(iterable, str):
+            for value in iterable:
+                self.push(value)
+        elif iterable:
+            raise TypeError
+            print("Please submit an iterable when instantiating the list. You can use the push method to add one value at a time.")
+
+    def push(self, val):
+        """Insert value to head of the list."""
+        new_node = Node(val, self.head)
+        self.head = new_node
+        self.size_of_list += 1
+
+    def pop(self):
+        """Remove value from head of list and return it to user."""
+        if self.head is None:
+            raise IndexError("Impossible to pop an empty list")
+        value_popped = self.head.val
+        self.head = self.head.next
+        self.size_of_list -= 1
+        return value_popped
+
+    def size(self):
+        """Return the size of the list."""
+        return self.size_of_list
+
+    def search(self, val):
+        """Return the val of the node when searched."""
+        current = self.head
+        while current:
+            if current.val == val:
+                return current
+            current = current.next
+        if current is None:
+            raise NameError
+            print("Parameter given, {}, is not in list".format(str(val)))
+        return current
```

Comment:
Why do you need lines 64 through 67? If search doesn't find what it's looking for, it should just return None. It's in the spec.

"search(val) will return the node containing ‘val’ in the list, if present, else None"